### DISCUSSION TITLE
PlayerBlockInteractionEvent event.setCanceled(true); won't cancel the event.

### DISCUSSION THREAD

**Discussion by xSehrMotiviert** [Minestom/Minestom#1425] [Help] [answered]
Hi,

my Problem here today is, that players still can open and close doors, trapdoors, fence gates and they can still press buttons and modify repeaters and comparators. (Only client side) I tested it with 2 accounts. If one changes something, the other one won't see it. 
But I want the player who changes something to see, that his changes will be reverted to the original state. (Door closed, Button not pressed, etc..)

The code that I have is
`public class PlayerBlockInteractEventHandler {

    public static void handle(@NotNull PlayerBlockInteractEvent event, InstanceContainer instanceContainer) {

        Player player = event.getPlayer();
        Block block = event.getBlock();

        System.out.println(block);

        event.setCancelled(true);
        MinecraftServer.getSchedulerManager().scheduleTask(() -> {
            player.sendPacket(new BlockChangePacket(event.getBlockPosition(), block.stateId()));
        }, TaskSchedule.nextTick(), TaskSchedule.stop());

        System.out.println(block);

    }
}`

**Comment by MrGazdag** [comment, marked as answer]
If you want to use `PlayerBlockInteractEvent` to cancel the interaction with a block, use `PlayerBlockInteractEvent#setBlockingItemUse(boolean)`. This will refresh the block for the player. Keep in mind, player movement checks are not done currently, so for example: if a player opens a door, and usage is blocked, they will still have enough time to slightly get into the block, before the block is synced to the player again, which will allow the player to go straight through the closed door. Keep this in mind while doing stuff, or do movement checks manually.

**Reply by xSehrMotiviert** [reply]
🤔 how would you do the movement check? I don't care at the moment, if the user goes through the door. I can always place a barrier block behind it to block the user from entering a house or so.

**Reply by MrGazdag** [reply]
Great question, Minestom probably stores the collision boxes of the blocks, you can try to find something about that, and check if the player collides with the collision box of the current block.
